src/Traits/Statable.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Sebdesign\SM\Traits;
+
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachine;
+use Illuminate\Database\Eloquent\Model;
+
+trait Statable
+{
+    /**
+     * @var StateMachine
+     */
+    protected $stateMachine;
+
+    public function history()
+    {
+        if ($this->isEloquent()) {
+            return $this->hasMany(self::HISTORY_MODEL_NAME);
+        }
+
+        /** @var Model $model */
+        $model = app(self::HISTORY_MODEL_NAME);
+
+        return $model->where(self::HISTORY_MODEL_FOREIGN_KEY, $this->{self::PRIMARY_KEY});
+    }
+
+    public function addHistoryLine(array $transitionData)
+    {
+        $transitionData['user_id'] = auth()->id();
```

Comment:
The user should not be hardcoded. Developers should be able to pass any `$transitionData` payload from outside.